telemetry-core/src/main/java/com/newrelic/telemetry/util/IngestWarnings.java

```diff
@@ -1,67 +1,44 @@
 package com.newrelic.telemetry.util;
 
+import com.newrelic.telemetry.Telemetry;
+import com.newrelic.telemetry.events.Event;
+import com.newrelic.telemetry.logs.Log;
+import com.newrelic.telemetry.metrics.Metric;
 import java.util.Map;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class IngestWarnings {
   private static final Logger logger = LoggerFactory.getLogger(IngestWarnings.class);
   private static final int MAX_NUMBER_OF_EVENT_ATTRIBUTES =
-      254; // Public documentation says 255, but true maximum is 254
+      254; // Public documentation says 255, but true maximum is 254. This is referenced in the
+  // following Slack thread: https://newrelic.slack.com/archives/CAHBUB7A9/p1625002090399400.
   private static final int MAX_NUMBER_OF_METRIC_ATTRIBUTES = 100;
   private static final int MAX_NUMBER_OF_LOG_ATTRIBUTES =
-      254; // Public documentation says 255, but true maximum is 254
+      254; // Public documentation says 255, but true maximum is 254. This is reference in the
+  // following Slack thread: https://newrelic.slack.com/archives/CAHBUB7A9/p1625002090399400
 
   private static final int MAX_ATTRIBUTE_NAME_LENGTH = 255;
   private static final int MAX_ATTRIBUTE_VALUE_LENGTH = 4096;
 
-  public void eventWarningNumAttributes() {
-    logger.warn(
-        "The number of attributes in this event is greater than the maximum allowed attributes per event.");
-  }
-
-  public void metricWarningNumAttributes() {
-    logger.warn(
-        "The number of attributes in this metric is greater than the maximum allowed attributes per metric.");
-  }
-
-  public void logWarningNumAttributes() {
-    logger.warn(
-        "The number of attributes in this log is greater than the maximum allowed attributes per log.");
-  }
-
-  public void attributeNameWarning(String attributeName) {
-    logger.warn(
-        "The length of the attribute named "
-            + attributeName
-            + " is greater than the maximum length allowed for an attribute name.");
-  }
-
-  public void attributeValueWarning(String attributeValue) {
-    logger.warn(
-        "The value of the attribute "
-            + attributeValue
-            + " is greater than the maximum length allowed for an attribute value.");
-  }
-
-  public void raiseIngestWarnings(Map<String, Object> attributes, String dataType) {
+  public void raiseIngestWarnings(Map<String, Object> attributes, Telemetry dataType) {
 
     // First Check - Check for valid number of attributes
 
     int numberOfAttributes = attributes.size();
-    if (dataType.equals("Event")) {
+    if (dataType instanceof Event) {
       if (numberOfAttributes > MAX_NUMBER_OF_EVENT_ATTRIBUTES) {
-        eventWarningNumAttributes();
+        warningNumAttributes("Event");
```

Comment:
Nice, looking better! 

One more tweak, could you try this: 

warningNumAttributes(dataType.getClass().getSimple())